editor/src/components/canvas/dom-lookup.ts

```diff
@@ -165,32 +186,103 @@ export function findFirstParentWithValidElementPathInner(
         target.parentElement,
         allowDescendantsFromPath,
         parentSceneValidPathsCache,
+        metadata,
+        point,
       )
     }
   }
 }
 
+function extendValidPathsWithNonDomParent(
+  domPath: {
+    static: string
+    staticPath: ElementPath
+    dynamic: ElementPath
+  },
+  nonDomPath: ElementPath,
+): ElementPath | null {
+  // check if there is a dom path which is a descendant of a non-dom path: then the
+  // non-dom path should be a valid target too
+  if (EP.isDescendantOfOrEqualTo(domPath.staticPath, nonDomPath)) {
+    const depthDiff = EP.fullDepth(domPath.staticPath) - EP.fullDepth(nonDomPath)
+    return EP.nthParentPath(domPath.dynamic, depthDiff)
+  }
+  return null
+}
+
+function extendValidPathsWithNonDomChild(
+  domPath: {
+    static: string
+    staticPath: ElementPath
+    dynamic: ElementPath
+  },
+  nonDomPath: ElementPath,
+  point: CanvasPoint,
+  dynamicElementPaths: ElementPath[],
+  metadata: ElementInstanceMetadataMap,
+): ElementPath | null {
+  // check if there is a non-dom path which has a frame which contains the mouse point
+  // and which is a direct child of a dom path
```

Comment:
actually, do we need the direct child restriction here? shouldn't it be enough that the non-dom path is a descendant of the last dom element without any dom elements in the middle? actually does it matter if we even filter for that? we always want to find the deepest valid path anyways.